internal/sdk/cloudian/qos.go

```diff
@@ -0,0 +1,137 @@
+package cloudian
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+)
+
+// QualityOfService configures data limits for a Group or User.
+type QualityOfService struct {
+	// Soft is the soft (warning) limit.
+	Soft QualityOfServiceLimits
+	// Hard is the hard limit.
+	Hard QualityOfServiceLimits
+}
+
+// QualityOfService configures data limits.
+type QualityOfServiceLimits struct {
+	// StorageQuotaKiBs is the limit for total stored data in KiB.
+	StorageQuotaKiBs *int64
+	// StorageQuotaCount is the limit for total number of objects.
+	StorageQuotaCount *int64
+	// RequestsPerMin is the limit for number of HTTP requests per minute.
+	RequestsPerMin *int64
+	// InboundKiBsPerMin is the limit for inbound data per minute in KiB.
+	InboundKiBsPerMin *int64
+	// OutboundKiBsPerMin is the limit for outbound data per minute in KiB.
+	OutboundKiBsPerMin *int64
+}
+
+// nolint: gocyclo
+func (qos *QualityOfService) unmarshalJSON(raw []byte) error {
```

Comment:
Should implement https://pkg.go.dev/encoding/json#Unmarshaler.

Reply:
My thinking was to not expose the internal method, as an sdk user should never need to do it. But we can change it.

Reply:
Ok, but then let's name the method something custom.